### PR TITLE
feat(rule): Require unique aria labels in checkboxgroup & radiogroup

### LINKED ISSUE #1111: False negative: 'group-labelledby' check in 'checkboxgroup' rule does not fail corner cases
Screen reader experience for the following corner cases is not as expected, hence the ```checkboxgroup``` rule should fail

- Case-1: Ungrouped checkboxes with same name and each having aria-labelledby pointing to a common label

- Case-2: Ungrouped checkboxes with explicit [label] and each having aria-labelledby pointing one of the checkbox labels

- Case-3: Ungrouped checkboxes with implicit [label] and each having aria-labelledby pointing one of the checkbox labels

**Link to live demo:** https://codepen.io/iamrafan/pen/MqjwOz

**AT test results:**
image

<pre><code>axe-core version: 3.0.3
axe-Coconut version: 3.4.0-alpha.2</code></pre>

### PR DESCRIPTION
**Pull request by WilcoFiers**
Update the group-labelledby check to enforce that in addition to having a shared label, radiobuttons and checkboxes with the same name also have a unique label. This impacts the checkboxgroup and radiogroup rules.

I merged in a fix from #1314, which changes the gruntfile. We should probably more that PR first.

Closes #1111

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [ ] Follows the commit message policy, appropriate for next version
- [ ] Has documentation updated, a DU ticket, or requires no documentation change
- [ ] Includes new tests, or was unnecessary
- [ ] Code is reviewed for security by: << Name here >>

### CODE REVIEW

File: lib/checks/forms/group-labelledby.js [hunk 1]
```diff
@@ -1,34 +1,61 @@
...
+
+if (uniqueLabels.length > 0 && sharedLabels.length > 0) {
+	this.data(data);
+	return true;
+}
+
+if (uniqueLabels.length > 0 && sharedLabels.length === 0) {
+	data.failureCode = 'no-shared-label';
+} else if (uniqueLabels.length === 0 && sharedLabels.length > 0) {
+	data.failureCode = 'no-unique-label';
```
Comment (dylanb) [inline lib/checks/forms/group-labelledby.js]: Why not always fail this even when there is no shared label?

**Comment by dylanb** [comment]
@WilcoFiers seems like the change to make @stephenmathieson a code owner broke our ability to approve changes in that code

**Comment by jkodu** [comment]
@dylanb PR - https://github.com/dequelabs/axe-core/pull/1331, fixed the codeowners issue.

**Comment by naltatis** [comment]
Feedback: Just updated axe an ran into this check. We finally could find the issue, we were having, but had to read through the tests and code modifications here to understand what was wrong.

This is the message we recieved
```
Ensures related <input type="radio"> elements have a group and that the group designation is consistent https://dequeuniversity.com/rules/axe/3.2/radiogroup?application=axeAPI

Fix any of the following:
  Elements with the name "segments-deliveryaddress-deliveryButtons" do not all have a unique label, referenced through aria-labelledby
  Element does not have a containing fieldset or ARIA group
  #billing-radiobutton
```

The actual problem was, that our radio buttons had a `aria-labelledby` that was pointing to the groups legend. Additionally the radios had an individual label (connected via for/id and label > input nesting), which we thought was sufficient. We were able to get rid of the error by expanding the aria annotiation by the radio inputs label-id `aria-labelledby="legend-id"` -> `aria-labelledby="legend-id label1-id"`.
The error message above and the linked documentation did not point to that.

Maybe you should consider improving the message.

**Comment by dylanb** [comment]
@naltatis Do you think you could create a new issue in this project that references this and also maybe suggest a text that might have helped you figure out what was going on?

**Pull request merged**